### PR TITLE
Makefile: add initial makefile with live-check command

### PR DESCRIPTION
**Pull request by zyga**
This patch adds a top-level makefile that contains just one command,
live-check, for running unit tests live, alongside your code editor.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# live-check observes all .go files outside of the vendor tree and whenever any
+# of them changes runs "go test ./..." which tests the whole tree. This is
+# handy to keep running in a separate window (on a 2nd monitor perhaps) while
+# editing code as it allows one to not have to switch to another application to
+# see the general status of unit tests.
+.PHONY: live-check
+live-check:
+	find -name "*.go" \! -path "./vendor/*" | entr -c go test ./...
```
Comment (bboozzoo) [inline Makefile]: Quite sure there's a community tool for this.

Comment (zyga) [reply Makefile]: Do you know the name of that tool?

**Comment by pedronis** [comment]
as a passerby having a Makefile with just this in would seem very strange

**Comment by pedronis** [comment]
removed simple because at least expressed as this Makefile this is a bit controversial

**Comment by zyga** [comment]
@pedronis yes, the plan is to grow this makefile to handle all of the build process (what @bboozzoo did essentially). Since we already agreed on this I thought that using a makefile for this would be natural.

File: Makefile (hunk 1 above)
Comment (bboozzoo) [reply Makefile]: There's https://github.com/go-godo/godo and it doesn't need a makefile (makes it less controversial?). OTOH it should be fairly simple to write something using `fsnotify`, with a configurable settle time and perhaps exploiting the lower level Go toolchain information.

**Comment by pedronis** [comment]
it's not natural in so far that it adds something by which "make" doesn't do any of the obvious things. given that it seems it's also unclear there is an obvious choice how to do it, it seems to belong to personal local setup.  ATM my vote on the whole thing is -1 in the makefile and -0 in general

**Pull request closed without merging**